Doc/library/contextlib.rst

```diff
@@ -177,12 +177,12 @@ Functions and classes provided:
       def process_file(file_or_path):
           if isinstance(file_or_path, str):
               # If string, open file
-              cm = open(file_or_path)
+              cm = open
           else:
-              # Caller is responsible for closing file
-              cm = nullcontext(file_or_path)
+              # Caller opened the file and is responsible for closing it
```

Comment:
Citing Nick's suggestion from the bpo:
> The documentation still needs updating to improve or remove the file-based example, and *point out that for resources that get acquired on construction, you either need to delay construction to the start of the with statement, or else use ExitStack instead of nullcontext*.

I think it's important to add that to the description, otherwise one might think this way of writing is just for the sake of shorter code. The reasoning behind this change is not clear at all, at first sight (I was surprised when I read https://bugs.python.org/issue34066)

Reply:
I agree, thank you.

Reply:
Actually after reading Nick's message more carefully I think there is more to this issue than I realized before. I'm going to close this PR.